src/Elastic.Apm/Metrics/MetricsCollector.cs

```diff
@@ -64,6 +66,19 @@ public MetricsCollector(IApmLogger logger, IPayloadSender payloadSender, IConfig
 
 			MetricsProviders = new List<IMetricsProvider>();
 
+			if (metricsProvider != null)
+			{
+				foreach (var item in metricsProvider)
+				{
+					if (item != null)
+					{
+						if (!WildcardMatcher.IsAnyMatch(currentConfigSnapshot.DisableMetrics, BreakdownMetricsProvider.SpanSelfTime))
```

Comment:
This makes the assumption that any added metrics providers should be controlled by the span self-time metric. Whilst this holds true now as breakdown provider is the only one added in the ctor, maybe the string(s) to match should be a property on an `IMetricsProvider` to allow each individual provider determine if it should be added.

Reply:
Yeah, I agree that this isn't ok.

> maybe the string(s) to match should be a property on an IMetricsProvider to allow each individual provider determine if it should be added.

Problem is we'd need to address 2 scenarios with this:
- The `BreakdownMetricsProvider` and everything passed to the `ctor` is already created - so yes, in that case we could just query the string(s) from `IMetricsProvider` and it'd indeed help.
- Every other `IMetricsProvider` is not instantiated at this point, so we can't query the string(s) list for those. In that cases what we do is that we check the `DisabledMetrics` list and if one of those is disabled we don't even instantiate those.

What I did is that I moved the check to `AgentCompontents` - that's where we instantiate the `BreakdownMetricsProvider`. 

Long term we could come up with an infrastructure to better handle this - I think it'd be fine to introduce the string list you suggest, but that'll also mean we'd need to instantiate all the `IMetricsProvider` and if they are not enabled based on the string(s) list, we'd just not add them to the list and let them GCed. Nevertheless that'd be a longer discussion, so maybe we could do that in a separate PR.

Reply:
Update: I decided to move on and implement it nevertheless. I added `IMetricsProvider.IsEnabled` - so we instantiate each `IMetricsProvider` implementation, but only add them if they are enabled. This changed a few things around metrics collection, but in my opinion this is now much cleaner.

Reply:
++ it'd be possible to build on this for the custom metrics scenarios.